udata_hydra/app.py

```diff
@@ -1,78 +1,74 @@
-    domain = fields.Str()
-    created_at = fields.DateTime()
-    check_status = fields.Integer(data_key="status")
-    headers = fields.Function(lambda obj: json.loads(obj["headers"]) if obj["headers"] else {})
-    timeout = fields.Boolean()
-    response_time = fields.Float()
-    error = fields.Str()
-    dataset_id = fields.Str()
-    resource_id = fields.UUID()
-    deleted = fields.Boolean()
-    parsing_started_at = fields.DateTime()
-    parsing_finished_at = fields.DateTime()
-    parsing_error = fields.Str()
-    parsing_table = fields.Str()
-
-
-class ResourceDocument(Schema):
-    id = fields.Str(required=True)
-    url = fields.Str(required=True)
-    format = fields.Str(allow_none=True)
-    title = fields.Str(required=True)
-    schema = fields.Dict(allow_none=True)
-    description = fields.Str(allow_none=True)
-    filetype = fields.Str(required=True)
-    type = fields.Str(required=True)
-    mime = fields.Str(allow_none=True)
-    filesize = fields.Int(allow_none=True)
-    checksum_type = fields.Str(allow_none=True)
-    checksum_value = fields.Str(allow_none=True)
-    created_at = fields.DateTime(required=True)
-    last_modified = fields.DateTime(required=True)
-    extras = fields.Dict()
-    harvest = fields.Dict()
-
-
-class ResourceQuery(Schema):
-    dataset_id = fields.Str(required=True)
-    resource_id = fields.Str(required=True)
-    document = fields.Nested(ResourceDocument(), allow_none=True)
-
-
 def _get_args(request, params=("url", "resource_id")) -> list:
+    """
+    Get GET parameters from request
+    """
     data = [request.query.get(param) for param in params]
     if not any(data):
         raise web.HTTPBadRequest()
     return data
 
 
+@routes.get("/api/resource/status/")
+async def get_resource_status(request: web.Request) -> web.Response:
+    """
+    Get the current status of a resource
+    """
+    try:
+        payload = await request.json()
+        valid_payload: Union[dict, list, None] = ResourceQuerySchema().load(payload)
+        if not isinstance(valid_payload, dict):
+            raise ValidationError("Invalid payload")
+    except ValidationError as err:
+        raise web.HTTPBadRequest(text=json.dumps(err.messages))
+
+    resource_id = valid_payload["resource_id"]
+
+    res = await Resource.get(resource_id)
+    if not res:
+        raise web.HTTPNotFound()
+    status: str = res[0]["status"]
+    verbose: str = Resource.STATUSES[status]
+
+    return web.json_response({"resource_id": resource_id, "status": status, "verbose": verbose})
+
+
 @routes.post("/api/resource/created/")
 async def resource_created(request: web.Request) -> web.Response:
+    """
+    Endpoint to receive a resource creation event from a source
+    Will create a new resource in the DB "catalog" table and mark it as priority for next crawling
+    Respond with a 200 status code and a JSON body with a message key set to "created"
+    If error, respond with a 400 status code
+    """
```

Comment:
Nit: [pep 257](https://peps.python.org/pep-0257/#multi-line-docstrings) recommends multi-line docstrings to be formatted as such:

```python
"""Endpoint to receive a resource creation event from a source.

Will create a new resource in the DB "catalog" table and mark it as priority for next crawling.
Respond with a 200 status code and a JSON body with a message key set to "created".
If error, respond with a 400 status code.
"""
```